packages/block-library/src/post-navigation-link/edit.js

```diff
@@ -47,6 +61,42 @@ export default function PostNavigationLinkEdit( {
 			[ `has-text-align-${ textAlign }` ]: textAlign,
 		} ),
 	} );
+
+	// We need to know the post type in order to get all supported taxonomies.
+	const postType = useSelect(
+		// FIXME: @wordpress/block-library should not depend on @wordpress/editor.
+		// Blocks can be loaded into a *non-post* block editor.
+		// eslint-disable-next-line @wordpress/data-no-store-string-literals
+		( select ) => select( 'core/editor' ).getCurrentPostType(),
+		[]
+	);
+	const taxonomies = useSelect( ( select ) => {
+		const { getTaxonomies } = select( coreStore );
+		const filteredTaxonomies = getTaxonomies( {
+			type: postType,
+			per_page: -1,
+			context: 'view',
+		} );
+		return filteredTaxonomies;
+	}, [] );
```

Comment:
```suggestion
	}, [ postType ] );
```

I think this hook has a dependency on `postType`, so it should be included in the dependency array.